lang/src/lib.rs

```diff
@@ -710,3 +723,39 @@ macro_rules! source {
         }
     };
 }
+
+#[test]
+fn test_instruction_data() {
+    // Define some test type and implement ser/de, discriminator, and ix data
+    #[derive(Default, borsh::BorshSerialize, borsh::BorshDeserialize, PartialEq, Eq)]
```

Comment:
Please use `AnchorSerialize` and `AnchorDeserialize` instead:

```suggestion
    #[derive(Default, AnchorSerialize, AnchorDeserialize, PartialEq, Eq)]
```